File: WEEK3/bj1_1920.py
import sys

def binary_search(arr, target):
    start, end = 0, len(arr) - 1
    while start <= end:
        mid = (start + end) // 2
        if arr[mid] == target:
            return 1
        elif arr[mid] < target:
            start = mid + 1
        else:
            end = mid - 1
    return 0

n = int(sys.stdin.readline())
arr_n = sorted(list(map(int, sys.stdin.readline().split())))
m = int(sys.stdin.readline())
arr_m = list(map(int, sys.stdin.readline().split()))

for target in arr_m:
    print(binary_search(arr_n, target))

File: WEEK3/bj1_1920.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const n = parseInt(input[0]);
const arrN = input[1].split(' ').map(Number).sort((a, b) => a - b);
const m = parseInt(input[2]);
const arrM = input[3].split(' ').map(Number);

function binarySearch(arr, target) {
    let start = 0;
    let end = arr.length - 1;
    while (start <= end) {
        let mid = Math.floor((start + end) / 2);
        if (arr[mid] === target) {
            return 1;
        } else if (arr[mid] < target) {
            start = mid + 1;
        } else {
            end = mid - 1;
        }
    }
    return 0;
}

const result = [];
for (let i = 0; i < m; i++) {
    result.push(binarySearch(arrN, arrM[i]));
}

console.log(result.join('\n'));
